### PR TITLE
Implements cetz diagrams #37

### PR DESCRIPTION
**Pull request by benabel**
This is a basic implementation using typst to compile cetz code to svg, png or pdf.

### CODE REVIEW

**Comment by benabel** [comment]
The filter works with pandoc, but recent changes in the filter makes it fail with quarto see #39.

**Review by tarleb — APPROVED** [review]
Looks great, thanks! The only request I have is about something minor in the docs :sweat_smile:

**Comment by tarleb** [comment]
Oh, and running the test during CI would be good, too. There seems to be a typst package available on Alpine, so using `pandoc/minimal:latest-alpine` as base image should work well.

File: README.md [hunk 1]
```diff
@@ -80,6 +80,7 @@ that can be used to specify a specific executable.
 | [Mermaid]   | `mermaid`         | `mmdc`     | `MERMAID_BIN`   |
 | [PlantUML]  | `plantuml`        | `plantuml` | `PLANTUML_BIN`  |
 | [Ti*k*Z]    | `tikz`            | `pdflatex` | `PDFLATEX_BIN`  |
+| [cetz]      | `cetz`            | `typst`    | `TYPST_BIN`     |
```
[the remark this answers is not in the document]
Comment (benabel) [reply README.md]: OK, done.

**Comment by benabel** [comment]
I've added the CI action using quarto image to use their built-in typst program.

**Comment by tarleb** [comment]
oh, that's a neat approach! It should be possible to set the executable in the metadata, something along the lines of

``` yaml
diagram:
  cetz:
    execpath: quarto typst
```

Then we won't have to do the "create executables" step that seems to cause issues on the GitHub workflow runners.

Let me know if that doesn't work, then I'll try to make it work somehow.

**Comment by benabel** [comment]
This approach do not work:

```
diagram/diagram.lua: quarto typst: createProcess: posix_spawnp: does not exist (No such file or directory
```

**Comment by tarleb** [comment]
OK, I'll try to implement a fix. One moment.

**Comment by tarleb** [comment]
Please rebase to the current main and try with `execpath: ['quarto', 'typst']`

**Comment by benabel** [comment]
Ok I try that.

**Comment by benabel** [comment]
I fixed the arguments list to use quarto typst command and changed pandoc command to use quarto's pandoc in makefile. This should be OK now.

**Comment by tarleb** [comment]
Thank you! I'll add the missing `make` install step.

**Comment by tarleb** [comment]
Works beautifully! I'll make a new release within the next days.

**Comment by benabel** [comment]
@tarleb Thanks for your feedbacks during the pr. And again thanks for sharing this handful extension open source.

**Pull request merged**